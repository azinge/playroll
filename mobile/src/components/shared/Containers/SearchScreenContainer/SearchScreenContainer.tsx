/**
 * SearchScreenContainer
 */

import * as React from 'react';
import Collapsible from 'react-native-collapsible-header';
import { isIphoneX } from 'react-native-iphone-x-helper';
import SearchScreenHeader from '../../Headers/SearchScreenHeader';
import { HeaderIconType } from '../../../../themes/Icons';
import { StyleProp, ViewStyle, RefreshControl, View } from 'react-native';

export interface HeaderProps {
  title?: string;
  icons?: HeaderIconType[];
  onChangeText?: (text: string) => void;
  onSubmitEditing?: (text: string) => void;
}

interface ContainerProps extends HeaderProps {
  renderHeader?: () => JSX.Element;
  renderSubHeader?: () => JSX.Element;
  flatList?: boolean;
  contentContainerStyle?: StyleProp<ViewStyle>;
  data?: any[];
  keyExtractor?: (item: any, i: number) => string;
  renderFlatListHeader?: () => JSX.Element;
  renderFlatListFooter?: () => JSX.Element;
  renderFlatListEmptyComponent?: () => JSX.Element;
  renderItem?: (obj: any) => any;
  refreshing?: boolean;
  onRefresh?: () => void;
  onEndReached?: () => void;
  onEndReachedThreshold?: number;
}

interface Props extends ContainerProps {}

interface State {}

export default class SearchScreenContainer extends React.Component<
  Props,
  State
> {
  getRefreshProps() {
    const { flatList, refreshing, onRefresh } = this.props;
    if (!onRefresh) {
      return {};
    }
    if (flatList) {
      return { refreshing, onRefresh };
    } else {
      return {
        refreshControl: (
          <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
        ),
      };
    }
  }
  render() {
    return (
      <Collapsible
        max={45}
        min={isIphoneX() ? 48 : 20}
        backgroundColor={'purple'}
        renderHeader={
          this.props.renderHeader
            ? this.props.renderHeader()
            : this.renderHeader()
        }
        renderContent={this.renderContent()}
        contentContainerStyle={this.props.contentContainerStyle}
        flatList={this.props.flatList}
        data={this.props.data}
        keyExtractor={this.props.keyExtractor}
        ListHeaderComponent={
          this.props.renderFlatListHeader
            ? this.props.renderFlatListHeader()
            : undefined
        }
        ListFooterComponent={
          this.props.renderFlatListFooter
            ? this.props.renderFlatListFooter()
            : undefined
        }
        ListEmptyComponent={
          this.props.renderFlatListEmptyComponent
            ? this.props.renderFlatListEmptyComponent()
            : undefined
        }
        renderItem={this.props.renderItem}
        onEndReached={this.props.onEndReached}
        onEndReachedThreshold={this.props.onEndReachedThreshold}
        {...this.getRefreshProps()}
      />
    );
  }
  renderHeader() {
    return (
      <SearchScreenHeader
        title={this.props.title}
        icons={this.props.icons}
        onChangeText={this.props.onChangeText}
        onSubmitEditing={this.props.onSubmitEditing}
      />
    );
  }
  renderContent() {
    return this.props.children;
  }
}
