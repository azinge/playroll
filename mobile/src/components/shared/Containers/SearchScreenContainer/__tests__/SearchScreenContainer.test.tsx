import * as React from "react";
import renderer from "react-test-renderer";

import SearchScreenContainer from "../SearchScreenContainer";

it("renders correctly", () => {
  const tree = renderer.create(<SearchScreenContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});
