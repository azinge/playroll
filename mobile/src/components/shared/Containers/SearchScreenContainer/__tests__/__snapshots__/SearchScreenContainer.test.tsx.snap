// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "purple",
        "height": 64,
        "left": 0,
        "paddingTop": 19,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "marginTop": -19,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "purple",
              "borderBottomColor": "#f2f2f2",
              "borderBottomWidth": 0.5,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "space-between",
              "paddingHorizontal": 10,
              "paddingTop": 20,
            }
          }
        >
          <Image
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": 64,
                  "width": undefined,
                },
                undefined,
              ]
            }
            testID="headerContainer"
          />
          <View
            style={
              Object {
                "alignItems": "flex-start",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 4,
                    "marginTop": 5,
                  }
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 3,
                "paddingHorizontal": 15,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#4F0C50",
                  "borderRadius": 5,
                  "flexDirection": "row",
                  "paddingHorizontal": 5,
                  "position": "relative",
                  "width": 275,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                onSubmitEditing={[Function]}
                placeholderTextColor="grey"
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 23,
                      "paddingLeft": 3,
                      "paddingTop": 1,
                    },
                    Object {
                      "color": "white",
                      "flex": 2,
                      "fontSize": 18,
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <View
                style={
                  Object {
                    "top": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "marginRight": 16,
                      "marginTop": 5,
                    },
                    Object {
                      "marginRight": 0,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                  underlayColor="rgba(255,255,255,0)"
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
